spec/percy/cli/snapshot_runner_spec.rb

```diff
@@ -99,8 +99,29 @@
       expect(paths).to match_array(expected_results)
     end
 
+    it 'returns only the files matching the snapshots_regex' do
+      opts = { snapshots_regex: 'xml' }
+      paths = runner._find_root_paths(root_dir, opts)
+
+      expected_results = [
+        File.join(root_dir, 'index.xml'),
+        File.join(root_dir, 'subdir/test.xml'),
+      ]
+
+      # Symlinked folders don't work out-of-the-box upon git clone on windows
+      unless Gem.win_platform?
+        expected_results += [
+          # Make sure directory symlinks are followed.
+          File.join(root_dir, 'subdir_symlink/test.xml'),
+        ]
+      end
+
+      expect(paths).to match_array(expected_results)
+    end
+
     it 'skips files passed into the ignore_regex option' do
-      paths = runner._find_root_paths root_dir, ignore_regex: 'skip|ignore'
+      opts = { ignore_regex: 'skip|ignore' }
+      paths = runner._find_root_paths(root_dir, opts)
```

Comment:
The bug that existed prior to this PR still doesn't have a test case around it. That's probably okay, but wanted to make it explicit.

If we did want to add one, I think it would involve testing the run function, and mocking out the things it calls, making sure the right params are received..    That might end up quite a time consuming test to right though, so the ROI might not be there.

Reply:
Yeah I did consider mocking that. I'll take a quick pass at getting that spec working to get an idea of how much work it will take.